### DISCUSSION TITLE
stabilizing profiles

### DISCUSSION THREAD

**Discussion by beroal** [roddhjav/apparmor.d#482] [General]
I see that `make enforce` install many profiles in complain mode. I guess it's because they are “unstable”. What is a procedure for stabilizing profiles? Say, you need testers?

**Comment by roddhjav** [comment]
There no strict procedure yet.

The initial plan was to ensure stable profile were all [tested](https://apparmor.pujol.io/development/integration/) against the supported distribution. However, my experiments regarding these tests were not very promising, therefore, at the moment these tests are mostly on pause. Ideally, the profiles should be tested against each distribution QA system (e.g. for opensuse https://openqa.opensuse.org/)
It is certainly not something I can do alone...

Other idea would be to split the package between core stable profile and the others. See #464 

Meanwhile in the past few months a lot of work has been done regarding the common abstraction used by the profile, and this itself solved a lot of compatibility issues.
